crypto/rand/rand_win.c

```diff
@@ -62,21 +88,27 @@ size_t rand_pool_acquire_entropy(RAND_POOL *pool)
         return entropy_available;
 # endif
 
-# ifdef USE_BCRYPTGENRANDOM
-    bytes_needed = rand_pool_bytes_needed(pool, 1 /*entropy_factor*/);
-    buffer = rand_pool_add_begin(pool, bytes_needed);
-    if (buffer != NULL) {
-        size_t bytes = 0;
-        if (BCryptGenRandom(NULL, buffer, bytes_needed,
+# if defined(USE_BCRYPTGENRANDOM) || defined(USE_BCRYPTGENRANDOM_DL)
+#  ifndef USE_BCRYPTGENRANDOM
+     RUN_ONCE(&bcrypt_init, do_bcrypt_init);
+#  endif
+    if (bcrypt_gen_random != NULL) {
+        bytes_needed = rand_pool_bytes_needed(pool, 1 /*entropy_factor*/);
+        buffer = rand_pool_add_begin(pool, bytes_needed);
+        if (buffer != NULL) {
+            size_t bytes = 0;
+            if (bcrypt_gen_random(NULL, buffer, bytes_needed,
                             BCRYPT_USE_SYSTEM_PREFERRED_RNG) == STATUS_SUCCESS)
-            bytes = bytes_needed;
+                bytes = bytes_needed;
 
-        rand_pool_add_end(pool, bytes, 8 * bytes);
-        entropy_available = rand_pool_entropy_available(pool);
+            rand_pool_add_end(pool, bytes, 8 * bytes);
+            entropy_available = rand_pool_entropy_available(pool);
+        }
+        if (entropy_available > 0)
+            return entropy_available;
     }
-    if (entropy_available > 0)
-        return entropy_available;
-# else
+# endif // USE_BCRYPTGENRANDOM || USE_BCRYPTGENRANDOM_DL
```

Comment:
We require traditional C comments in our source, so this line needs to be changed.

Reply:
updated